criu/cr-dump.c

```diff
@@ -846,8 +847,10 @@ static int dump_task_thread(struct parasite_ctl *parasite_ctl,
 				const struct pstree_item *item, int id)
 {
 	struct parasite_thread_ctl *tctl = dmpi(item)->thread_ctls[id];
+	struct parasite_dump_cgroup_args cgroup_args, *info = NULL;
```

Comment:
I haven't understood why the info pointer is needed. Even in the existing code. Why can't we just use &cgroup_args in place of info.

Reply:
For both the existing + added code, the distinction is what we're doing with that memory. 

For example, [the existing code calls &cgroup_args directly](https://github.com/checkpoint-restore/criu/blob/958b750a3d6f3bbebfe8cbc800952dc39587485e/criu/cr-dump.c#L779) when the parasite is involved. As the name `info` imples, the existing code calls info when non-parasitic code needs the contents of the cgroup_args struct. 

I believe it's a matter of manipulation vs. collection, and for consistency-sake, I followed how the existing code achieved this. I wouldn't have been able to figure this out without some of your *pointers* -- I'd be happy to explore this more if we need more clarification!

Reply:
I agree with @kashyapp

Reply:
please remove the `info` variable